crates/sui-graphql-rpc/src/types/sui_address.rs

```diff
@@ -56,3 +55,137 @@ impl SuiAddress {
+        assert_eq!(
+            parsed.0,
+            [
+                1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69,
+                103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239
+            ]
+        );
+    }
+
+    #[test]
+    fn test_to_value() {
+        let addr = SuiAddress([
+            1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103,
+            137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239,
+        ]);
+        let value = <SuiAddress as ScalarType>::to_value(&addr);
+        assert_eq!(
+            value,
+            Value::String(
+                "0x0123456789abcdef0123456789abcdef0123456789abcdef0123456789abcdef".to_string()
+            )
+        );
+    }
+
+    #[test]
+    fn test_into_array() {
+        let addr = SuiAddress([
+            1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103,
+            137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239,
+        ]);
+        let arr = addr.into_array();
+        assert_eq!(
+            arr,
+            [
+                1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69,
+                103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239
+            ]
+        );
+    }
+
+    #[test]
+    fn test_from_array() {
+        let arr = [
+            1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103,
+            137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239,
+        ];
+        let addr = SuiAddress::from_array(arr);
+        assert_eq!(
+            addr,
+            SuiAddress([
+                1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69,
+                103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239
+            ])
+        );
+    }
+
+    #[test]
+    fn test_as_slice() {
+        let addr = SuiAddress([
+            1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103,
+            137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239,
+        ]);
+        let slice = addr.as_slice();
+        assert_eq!(
+            slice,
+            &[
+                1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69,
+                103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239
+            ]
+        );
+    }
+
+    #[test]
+    fn test_round_trip() {
+        let addr = SuiAddress([
+            1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239, 1, 35, 69, 103,
+            137, 171, 205, 239, 1, 35, 69, 103, 137, 171, 205, 239,
+        ]);
+        let value = <SuiAddress as ScalarType>::to_value(&addr);
+        let parsed_back = <SuiAddress as ScalarType>::parse(value).unwrap();
+        assert_eq!(addr, parsed_back);
+    }
+
+    #[test]
+    fn test_parse_invalid_prefix() {
```

Comment:
Would a test for `1x24124[...]` make sense?

Reply:
oh that's a good one